File: scrapers/sources/ohmi.pl
{
	searchURL => 'http://www.ohmigallery.com/DB/sitemap.asp',
	searchStart => 0,
	searchEnd => 0,
	searchIncrement => 1,

	urlExtractor => scraper {
		process '//a', 'urls[]' => '@href';
	},

	cleanURL => sub {
		my $url = shift;
		return $url =~ /(\d+)$/ ?
			"http://www.ohmigallery.com/DB/ItemDetail.asp?item=$1" :
			"";
	},

	extractPageID => qr/(\d+)/,

	pageExtractor => scraper {
		process '//center/a[contains(@href,"LargeImage")]', source_image => '@href';
	},

	cleanPage => sub {
		my $item = shift;
		$item->{'source_image'} =~ s!.*Images!http://www.ohmigallery.com/DB/Images!;
		return $item;
	},

	extractImageFile => qr/([^\/]+.jpg)$/
}

File: scrapers/images/ohmi.js
exports.extract = {
	artist: ["//td[contains(font/text(), 'Artist')]/following-sibling::td[1]",
		function(text) {
			return text.replace(/ [\(-].*/, "");
		}],
	title: "//td[contains(font/text(), 'Title')]/following-sibling::td[1]",
	date: ["//td[contains(font/text(), 'Date')]/following-sibling::td[1]",
		function(text) {
			return text.replace(/^(\d+).*/, "$1");
		}],
	description: "//td[contains(font/text(), 'Notes')]/following-sibling::td[1]",
	source_image: ["//center/a[contains(@href,'LargeImage')]/@href",
		function(text) {
			return text.replace(/.*Images/, "http://www.ohmigallery.com/DB/Images");
		}]
};

exports.genURL = function( id ) {
	return "http://www.ohmigallery.com/DB/ItemDetail.asp?item=" + id;
};
